### PR TITLE
DFW manual functions

### PR DESCRIPTION
**Pull request by Physarah**
A set of functions to take manual observations for the dfw run.

To take science images:
```
from manual_functions import do_science_setup
import numpy as np

cam_name = 'camera.zwo.0'
filter_type = 'g'
initial_focus_position = 22266
target_name = 'FRB190711'

pocs_observation = do_science_setup(cam_name, 
                                   target_name, 
                                   initial_focus_position, 
                                   filter_type)

pocs_observation.observatory.observe()
```

### CODE REVIEW

File: scripts/dfw_manual_functions.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    scheduler = create_scheduler_from_config(config)
+    mount = create_mount_from_config(config)  
+    
+    chosen_camera = cameras[cam_name]
+    chosen_camera.filterwheel.move_to(filter_type)    
+
+    # Set to best focus position found
+    chosen_camera.focuser.move_to(initial_focus_position)    
+    
+    simulators=['weather', 'mount', 'power', 'night']
```
Comment (lspitler) [inline scripts/dfw_manual_functions.py:115]: move to arg of this function

File: scripts/dfw_manual_functions.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    observatory = HuntsmanObservatory(scheduler=scheduler, simulators=simulators, mount=mount)
+
+    pocs = POCS(observatory)    
+    pocs.observatory.add_camera(cam_name, chosen_camera)    
+    test_observation = pocs.observatory.scheduler.observations[target_name]
```
Comment (lspitler) [inline scripts/dfw_manual_functions.py:40]: rename `test_observations` as it isnt a test

File: scripts/dfw_manual_functions.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    pocs.observatory.current_observation = test_observation    
+    test_observation.exptime = exposure_time
+    pocs.observatory.take_flat_fields(camera_names = cam_name)
```
Comment (lspitler) [inline scripts/dfw_manual_functions.py:43]: This must be a list:
```suggestion
    pocs.observatory.take_flat_fields(camera_names = [cam_name])
```

File: scripts/dfw_manual_functions.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    
+    simulators=['weather', 'power', 'night']
+    observatory = HuntsmanObservatory(scheduler=scheduler, simulators=simulators, mount = mount)
+
+    pocs = POCS(observatory)    
+    pocs.observatory.add_camera(cam_name, chosen_camera)    
+    test_observation = pocs.observatory.scheduler.observations[target_name]
+    pocs.observatory.current_observation = test_observation    
+    test_observation.exptime = exposure_time
+    pocs.observatory.take_dark_fields(exptimes = exposure_time)
```
Comment (lspitler) [inline scripts/dfw_manual_functions.py:70]: Do you need to add the arg: `camera_names = [cam_name]`?

File: scripts/dfw_manual_functions.py (hunk 1 above)
Comment (lspitler) [inline scripts/dfw_manual_functions.py:115]: make this an arg to `do_science_setup`

File: scripts/dfw_manual_functions.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    scheduler = create_scheduler_from_config(config)
+    mount = create_mount_from_config(config)  
+    
+    chosen_camera = cameras[cam_name]
+    chosen_camera.filterwheel.move_to(filter_type)    
+
+    # Set to best focus position found
+    chosen_camera.focuser.move_to(initial_focus_position)    
+    
+    simulators=['weather', 'mount', 'power', 'night']
```
Comment (lspitler) [inline scripts/dfw_manual_functions.py:88]: make arg to `do_science`

File: scripts/dfw_manual_functions.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    mount = create_mount_from_config(config)
+    mount.initialize()
+    
+    chosen_camera = cameras[cam_name]
+    chosen_camera.filterwheel.move_to(filter_type)    
+
+    # Set to best focus position found
+    chosen_camera.focuser.move_to(initial_focus_position)    
+    
+    simulators=['weather', 'power', 'night']
```
Comment (lspitler) [inline scripts/dfw_manual_functions.py:62]: make arg

**Review by lspitler — CHANGES_REQUESTED** [review]
@Physarah looking good. Not yet approving so can keep tweaking based upon improvements during the week. Can you incorporate what you learned last night into this file?

Also, it might just make sense to make this a script by adding the following at the bottom of the file:

```python

def main(obstype,
                cam_name, 
                target_name, 
                initial_focus_position, 
                filter_type,
                num_sciobs=1,
                ...):
    if obstype is "sci":
        pocs_observation = do_science_setup(cam_name, 
                                   target_name, 
                                   initial_focus_position, 
                                   filter_type)
        for i in range(num_sciobs):
             pocs_observation.observatory.observe()
   elif obstype is "dark":
        stuff
   elif obstype is "flat":
        stuff

if __name__ == '__main__':
    parser = argparse.ArgumentParser()
    parser.add_argument('--obs-type', type=str, help='Options: sci, dark, flat.', choices=["sci", "dark", "flat"])
    parser.add_argument('--cam-name', type=str, help='Camera name in config.yaml to use.')
    parser.add_argument('--filter', type=str, help='Valid filter name.')
    parser.add_argument('--focus', type=int, help='The camera's best focus value.')
    parser.add_argument('--target', type=str, help='Target to observe in targets.yaml.')
    parser.add_argument('--num-sciobs', type=str, help='Number of science exposures to take.')
    parser.add_argument('--exptime', type=str, help='Override default exptime.')

    args = parser.parse_args()
    obstype = args.obstype

    main(meta_db_location)


```

**Comment by wtgee** [comment]
> @Physarah looking good. Not yet approving so can keep tweaking based upon improvements during the week. Can you incorporate what you learned last night into this file?
> 
> Also, it might just make sense to make this a script by adding the following at the bottom of the file:
> 
> ```
> python
> 
> def main(obstype,
>                 cam_name, 
>                 target_name, 
>                 initial_focus_position, 
>                 filter_type,
>                 num_sciobs=1,
>                 ...):
>     if obstype is "sci":
>         pocs_observation = do_science_setup(cam_name, 
>                                    target_name, 
>                                    initial_focus_position, 
>                                    filter_type)
>         for i in range(num_sciobs):
>              pocs_observation.observatory.observe()
>    elif obstype is "dark":
>         stuff
>    elif obstype is "flat":
>         stuff
> 
> if __name__ == '__main__':
>     parser = argparse.ArgumentParser()
>     parser.add_argument('--obs-type', type=str, help='Options: sci, dark, flat.', choices=["sci", "dark", "flat"])
>     parser.add_argument('--cam-name', type=str, help='Camera name in config.yaml to use.')
>     parser.add_argument('--filter', type=str, help='Valid filter name.')
>     parser.add_argument('--focus', type=int, help='The camera's best focus value.')
>     parser.add_argument('--target', type=str, help='Target to observe in targets.yaml.')
>     parser.add_argument('--num-sciobs', type=str, help='Number of science exposures to take.')
>     parser.add_argument('--exptime', type=str, help='Override default exptime.')
> 
>     args = parser.parse_args()
>     obstype = args.obstype
> 
>     main(meta_db_location)
> ```

We want to start writing scripts with [`click`](https://click.palletsprojects.com/en/7.x/) so it's good to get in the habit.  It's significantly easier.

**Comment by lspitler** [comment]
> 
> We want to start writing scripts with [`click`](https://click.palletsprojects.com/en/7.x/) so it's good to get in the habit. It's significantly easier.

Ha, thanks for pointing this out. Looks useful.

**Comment by lspitler** [comment]
@Physarah please push any and all changes you've implemented so far and ping me for review. thanks!

**Pull request closed without merging**